light/beacon/api/light_api.go

```diff
@@ -0,0 +1,372 @@
+	url           string
+	client        *http.Client
+	customHeaders map[string]string
+}
+
+func NewBeaconLightApi(url string, customHeaders map[string]string) *BeaconLightApi {
+	return &BeaconLightApi{
+		url: url,
+		client: &http.Client{
+			Timeout: time.Second * 10,
+		},
+		customHeaders: customHeaders,
+	}
+}
+
+func (api *BeaconLightApi) httpGet(path string) ([]byte, error) {
+	req, err := http.NewRequest("GET", api.url+path, nil)
+	if err != nil {
+		return nil, err
+	}
+	for k, v := range api.customHeaders {
+		req.Header.Set(k, v)
+	}
+	resp, err := api.client.Do(req)
+	if err != nil {
+		return nil, err
+	}
+	defer resp.Body.Close()
+	return io.ReadAll(resp.Body)
+}
+
+// Header defines a beacon header and supports JSON encoding according to the standard beacon API format
+type jsonHeader struct {
+	Slot          common.Decimal `json:"slot"`
+	ProposerIndex common.Decimal `json:"proposer_index"`
+	ParentRoot    common.Hash    `json:"parent_root"`
+	StateRoot     common.Hash    `json:"state_root"`
+	BodyRoot      common.Hash    `json:"body_root"`
+}
+
+func (h *jsonHeader) header() beacon.Header {
+	return beacon.Header{
+		Slot:          uint64(h.Slot),
+		ProposerIndex: uint(h.ProposerIndex),
+		ParentRoot:    h.ParentRoot,
+		StateRoot:     h.StateRoot,
+		BodyRoot:      h.BodyRoot,
+	}
+}
+
+// GetBestUpdateAndCommittee fetches and validates LightClientUpdate for given period and full serialized
+// committee for the next period (committee root hash equals update.NextSyncCommitteeRoot).
+// Note that the results are validated but the update signature should be verified by the caller as its
+// validity depends on the update chain.
+func (api *BeaconLightApi) GetBestUpdateAndCommittee(period uint64) (beacon.LightClientUpdate, []byte, error) {
+	resp, err := api.httpGet("/eth/v1/beacon/light_client/updates?start_period=" + strconv.Itoa(int(period)) + "&count=1")
+	if err != nil {
+		return beacon.LightClientUpdate{}, nil, err
+	}
+
+	type committeeUpdate struct {
+		Header                  jsonHeader          `json:"attested_header"`
+		NextSyncCommittee       syncCommitteeJson   `json:"next_sync_committee"`
+		NextSyncCommitteeBranch beacon.MerkleValues `json:"next_sync_committee_branch"`
+		FinalizedHeader         jsonHeader          `json:"finalized_header"`
+		FinalityBranch          beacon.MerkleValues `json:"finality_branch"`
+		Aggregate               syncAggregate       `json:"sync_aggregate"`
+	}
+
+	var data struct {
+		Data []committeeUpdate `json:"data"`
+	}
+	if err := json.Unmarshal(resp, &data); err != nil {
+		return beacon.LightClientUpdate{}, nil, err
+	}
+	if len(data.Data) != 1 {
+		return beacon.LightClientUpdate{}, nil, errors.New("invalid number of committee updates")
+	}
+	c := data.Data[0]
+	if len(c.NextSyncCommittee.Pubkeys) != 512 {
+		return beacon.LightClientUpdate{}, nil, errors.New("invalid number of pubkeys in next_sync_committee")
+	}
+
+	committee, ok := c.NextSyncCommittee.serialize()
```

Comment:
We should also check if the responded update is matched with period.

It should be very simple

`c.Header.Period  == period`

Reply:
done